src/components/structures/MatrixChat.js

```diff
@@ -364,7 +364,7 @@ module.exports = React.createClass({
                 // is completed in another browser, we'll be 401ed for using
                 // a guest access token for a non-guest account.
                 // It will be restarted in onReturnToGuestClick
-                Lifecycle.stopMatrixClient();
+                Lifecycle.stopMatrixClient(false);
```

Comment:
I'm not asking you to fix up anything - I'm just concerned that by changing this behaviour other parts of the app will break. This PR is titled "Call MatrixClient.clearStores **on logout**" : so changing the behaviour of upgrade registration feels highly undesirable. Why do we need to change the behaviour at all?